client/pkg/omni/resources/omni/machine_set.go

```diff
@@ -73,3 +73,16 @@ func (MachineSetExtension) ResourceDefinition() meta.ResourceDefinitionSpec {
 		PrintColumns:     []meta.PrintColumn{},
 	}
 }
+
+// GetMachineAllocation from the machine set resource.
+func GetMachineAllocation(res *MachineSet) *specs.MachineSetSpec_MachineAllocation {
+	if res.TypedSpec().Value.MachineClass != nil {
+		return res.TypedSpec().Value.MachineClass
```

Comment:
Should we mark this and the other field as `Deprecated:` and tell that the right way of getting allocation is using this function instead? The same way I did in compressed resources.

Reply:
Also, I'm thinking - what if we reduced this into a single field instead, and add another boolean or enum to indicate whether it's a class or request set? Or you think this is a clearer way?

Reply:
> Should we mark this and the other field as Deprecated: and tell that the right way of getting allocation is using this function instead? The same way I did in compressed resources.

Well, this method is internal to our code base. And we don't deprecate `MachineClass` usage.

> Also, I'm thinking - what if we reduced this into a single field instead, and add another boolean or enum to indicate whether it's a class or request set? Or you think this is a clearer way?

I had the same thoughts. But maybe later I'll add more fields to the `MachineRequestSet` allocation time. I'm thinking to put the config for machine request set scaling.

Reply:
> Also, I'm thinking - what if we reduced this into a single field instead, and add another boolean or enum to indicate whether it's a class or request set? Or you think this is a clearer way?

Actually after stitching more pieces together I'm starting to lean towards the flag in the single allocation config field.

Reply:
as you wish, you know the internals better here. I was just pointing out in case you haven't though of that option.